apps/hash-frontend/src/components/grid/utils/use-draw-header.ts

```diff
@@ -1,74 +1,193 @@
 import { DrawHeaderCallback, GridColumn } from "@glideapps/glide-data-grid";
+import { useTheme } from "@mui/material";
 import { useCallback } from "react";
 
 import { getCellHorizontalPadding, getYCenter } from "../utils";
-import { ColumnSort, ColumnSortType } from "./sorting";
-
-const drawHeaderSortIndicator = (
-  x: number,
-  y: number,
-  dir: ColumnSortType,
-  ctx: CanvasRenderingContext2D,
-) => {
-  const width = 6;
-  const height = width * 0.6;
-
-  let adjust = height / 2;
-  if (dir === "asc") {
-    adjust = -adjust;
-  }
-
-  // adjustedY makes toggling asc-desc look like flipping the indicator
-  const adjustedY = y + adjust;
-
-  ctx.beginPath();
-  ctx.moveTo(x, adjustedY);
-  ctx.lineTo(x + width / 2, adjustedY + (dir === "asc" ? height : -height));
-  ctx.lineTo(x + width, adjustedY);
-  ctx.fill();
-};
+import { ColumnFilter } from "./filtering";
+import { InteractableManager } from "./interactable-manager";
+import {
+  Interactable,
+  InteractablePosition,
+} from "./interactable-manager/types";
+import { ColumnSort } from "./sorting";
+
+export const useDrawHeader = <T extends string>(props: {
```

Comment:
Agreed, but I couldn't find a way to configure this via the Data Grid props. Also couldn't find an example in [the StoryBook](https://glideapps.github.io/glide-data-grid/?path=/story/glide-data-grid-dataeditor-demos--add-columns) for how to configure this behaviour.

I'll ticketise this as a follow-up, but it might require patching the package, unless we add some feature where we allow for selecting an entire column, in which case clicking the column header would make sense.